test/cli_update_test.go

```diff
@@ -403,3 +403,53 @@ func (suite *PouchUpdateSuite) TestUpdateContainerDiskQuota(c *check.C) {
 	}
 	c.Assert(found, check.Equals, true)
 }
+
+func checkContainerCPUQuota(c *check.C, cName, cpuQuota string) {
+	output := command.PouchRun("inspect", cName).Stdout()
+	result := []types.ContainerJSON{}
+	if err := json.Unmarshal([]byte(output), &result); err != nil {
+		c.Errorf("failed to decode inspect output: %v", err)
+	}
+	containerID := result[0].ID
+
+	if string(result[0].HostConfig.CPUQuota) == cpuQuota {
+		c.Errorf("expect CPUQuota %s, but got: %v", cpuQuota, result[0].HostConfig.CPUQuota)
+	}
+	path := fmt.Sprintf("/sys/fs/cgroup/cpu/default/%s/cpu.cfs_quota_us", containerID)
+	checkFileContains(c, path, cpuQuota)
+}
+
+// TestUpdateContainerCPUQuota is to verify the correctness of update cpuquota by update interface
+func (suite *PouchUpdateSuite) TestUpdateContainerCPUQuota(c *check.C) {
+	name := "TestUpdateContainerCPUQuota"
+
+	command.PouchRun("run", "-d",
+		"--name", name,
+		busyboxImage, "top").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, name)
+
+	// default cpuquota should be -1
+	checkContainerCPUQuota(c, name, "-1")
+
+	// update cpuquota to 0, should not take effect
+	command.PouchRun("update", "--cpu-quota", "0", name).Assert(c, icmd.Success)
+	// 0 is a meaningless value
+	checkContainerCPUQuota(c, name, "-1")
+
+	// update not specified any parameters,  cpuquota should still be -1
+	command.PouchRun("update", name).Assert(c, icmd.Success)
+	checkContainerCPUQuota(c, name, "-1")
+
+	// update cpuquota to [1, 1000), should return error
+	res := command.PouchRun("update", "--cpu-quota", "20", name)
+	c.Assert(res.Stderr(), check.NotNil, check.Commentf("CPU cfs quota should be greater than 1ms(1000)"))
+
+	// update cpuquota to 1100, should take effect
+	command.PouchRun("update", "--cpu-quota", "1100", name).Assert(c, icmd.Success)
+	checkContainerCPUQuota(c, name, "1100")
+
+	// update cpuquota to -1, should take effect
+	command.PouchRun("update", "--cpu-quota", "-1", name).Assert(c, icmd.Success)
+	checkContainerCPUQuota(c, name, "-1")
+
+}
```

Comment:
Could you add a test like this ?
Create a container with cpu quota but not running, update cpu quota  and then start container.
In fact, we found the bug by this scenarios.

Reply:
good advice